tslint.json

```diff
@@ -0,0 +1,74 @@
+{
+  "extends": [
+    "tslint:latest",
+    "tslint-react",
+    "tslint-eslint-rules",
+    "tslint-config-prettier",
+    "tslint-plugin-prettier"
+  ],
+  "rules": {
+    "no-var-requires": false,
+    "no-implicit-dependencies": false,
+    "ordered-imports": false,
+    "object-literal-sort-keys": false,
+    "ter-arrow-parens": [false],
+    "variable-name": [false],
+    "trailing-comma": false,
+    "no-console": true,
+    "triple-equals": [true, "allow-null-check", "allow-undefined-check"],
+    "whitespace": [false],
+    "ter-no-mixed-spaces-and-tabs": {
+      "type": "tabs"
+    },
+    "no-empty": [true, "allow-empty-catch"],
+    "no-inner-declarations": [true, "both"],
+    "valid-jsdoc": [
+      false,
+      {
+        "requireReturn": false,
+        "requireParamDescription": false
+      }
+    ],
+    "no-consecutive-blank-lines": [true],
+    "no-trailing-whitespace": [true, "ignore-blank-lines"],
+    "no-unused-variable": [
+      true,
+      "check-parameters",
+      {
+        "ignore-pattern": "^_"
+      }
+    ]
+  },
+  "no-use-before-declare": false,
+  "quotemark": [false],
```

Comment:
The [eslint-plugin-test-names](https://github.com/DanielMSchmidt/eslint-plugin-test-names) rule is missing, I would really love to have it added. Currently we have no tests written in TS, but that should change in the future

Reply:
The reason we introduced TS quickly is to allow us to move forward with a better-written data layer. Having a lint rule to verify tests do not contain "should" is a great thing, but does not really help with that goal. 

I would propose either to be added together with the first test written in .ts, or to be done when we unify the two tslint.config files into a single repo.

We should move forward here without that rule.

Reply:
@DanielMSchmidt please take a look at my review for more info on tslint rules. https://github.com/dcos/dcos-ui/pull/2854#discussion_r176583067

Reply:
I think a part of introducing typescript needs to be having the same coding standards enforced in typescript and in javascript. I know the config is based on the one in the ui-kit, it was my bad that I did not spot it there. The only alternative for me would be to have a new JIRA ticket for having a docs-labs/tslint-config repo, so that we are on the same page here. Adding them on the fly when we need them is IMHO not the best approach when it comes to linting (I know we won't come in this situation, but enforcing new standards at a later point of a project really is a pain, [see this ever stale PR](https://github.com/wix/detox/pull/517))

Reply:
As long as it is not blocking us delivering great things, I agree we could take a slower but higher quality approach. I would give +1 to that, and try to get this ticket in the next sprint. I am ok waiting for this to get typescript in.

Reply:
I wouldn't call 10 LOC to be copied over blocking, TBH.
For me we can clean up afterwards, but I want to make sure we have a ticket for that in the next sprint

Reply:
@DanielMSchmidt we are aware of that and that's the goal, to have a single source of truth for the tslint config.

**One thing that we need to keep in mind** is that tslint and eslint don't have the same rules and that's a fact nothing we can do about it besides trying to collaborate to tslint project adding the missing rules.

Reply:
Oh I understood https://github.com/buzinas/tslint-eslint-rules like it could "polyfill" the same interface, but I looked it up and it doesn't work that way. If we start writing tests in TS (what I would like us to do) I will just write a tslint plugin for the same thing :)